### PR TITLE
Update Vue dependencies

### LINKED ISSUE #13466: Vue dependencies need updating in 7.0 beta 0
##### **Overview of the issue**

If I create a new reactive Vue monolith and run `ncu` on its `package.json`, there's a lot of dependencies that need updating:

```
 @openapitools/openapi-generator-cli  1.0.13-4.3.1  →    2.1.16
 axios                                      0.21.0  →    0.21.1
 bootstrap-vue                              2.21.1  →    2.21.2
 dayjs                                       1.9.7  →    1.10.3
 swagger-ui-dist                            3.25.4  →    3.39.0
 vue-i18n                                   8.22.2  →    8.22.3
 vue-property-decorator                      9.0.2  →     9.1.2
 vue2-filters                               0.11.0  →    0.11.1
 vuex                                        3.5.1  →     3.6.0
 @types/jest                               26.0.10  →   26.0.20
 @types/node                                14.0.4  →  14.14.20
 @types/sinon                                9.0.8  →    9.0.10
 @typescript-eslint/eslint-plugin            4.7.0  →    4.13.0
 @typescript-eslint/parser                   4.7.0  →    4.13.0
 @vue/cli-plugin-eslint                      4.5.8  →    4.5.10
 @vue/cli-plugin-typescript                  4.5.8  →    4.5.10
 @vue/cli-service                            4.5.8  →    4.5.10
 @vue/test-utils                             1.1.1  →     1.1.2
 autoprefixer                               10.0.1  →    10.2.1
 browser-sync-webpack-plugin                 2.2.2  →     2.3.0
 copy-webpack-plugin                         6.3.0  →     7.0.0
 css-loader                                  3.5.3  →     5.0.1
 cypress                                     6.1.0  →     6.2.1
 eslint                                     7.13.0  →    7.17.0
 eslint-plugin-prettier                      3.1.4  →     3.3.1
 eslint-plugin-vue                           6.2.2  →     7.4.1
 fork-ts-checker-webpack-plugin              4.1.6  →     6.1.0
 html-webpack-plugin                         4.5.0  →     4.5.1
 husky                                       4.3.6  →     4.3.7
 jest                                       26.4.2  →    26.6.3
 jest-junit                                 11.1.0  →    12.0.0
 mini-css-extract-plugin                     0.9.0  →     1.3.3
 node-notifier                               8.0.0  →     9.0.0
 postcss                                     8.1.6  →     8.2.4
 postcss-import                             13.0.0  →    14.0.0
 postcss-loader                              4.0.4  →     4.1.0
 postcss-url                                10.1.0  →    10.1.1
 sass                                       1.29.0  →    1.32.4
 sass-loader                                10.0.5  →    10.1.1
 sinon                                       9.2.1  →     9.2.3
 terser-webpack-plugin                       4.2.3  →     5.1.1
 ts-jest                                    26.2.0  →    26.4.4
 ts-loader                                   7.0.4  →    8.0.14
 tslib                                       2.0.0  →     2.1.0
 wait-on                                     5.2.0  →     5.2.1
 webpack                                    4.44.2  →    5.13.0
 webpack-cli                                3.3.11  →     4.3.1
 webpack-dev-server                         3.11.0  →    3.11.1
 webpack-merge                               5.3.0  →     5.7.3
```

##### **Motivation for or Use Case**

Updating dependencies is usually a good idea for security reasons.

##### **Reproduce the error**

Create an app with the following JDL, `npm i npm-check-updates`, then run `ncu`.

```
application {
  config {
    baseName blog
    applicationType monolith
    authenticationType oauth2
    packageName com.jhipster.demo.blog
    prodDatabaseType mysql
    cacheProvider hazelcast
    buildTool maven
    clientFramework vue
    reactive true
    testFrameworks [cypress]
  }
  entities *
}

entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}

entity Post {
  title String required
  content TextBlob required
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}

relationship ManyToMany {
  Post{tag(name)} to Tag{entry}
}

paginate Post, Tag with infinite-scroll
```

##### **Suggest a Fix**

Update dependencies.

##### **JHipster Version(s)**

```
blog@0.0.0 /Users/mraible/dev/choose-mono
└── generator-jhipster@7.0.0-beta.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "authenticationType": "oauth2",
    "cacheProvider": "no",
    "clientFramework": "vue",
    "serverPort": "8080",
    "serviceDiscoveryType": false,
    "skipUserManagement": true,
    "withAdminUi": true,
    "baseName": "blog",
    "buildTool": "maven",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": ["en"],
    "messageBroker": false,
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "skipClient": false,
    "testFrameworks": ["cypress"],
    "websocket": false,
    "applicationType": "monolith",
    "packageName": "com.jhipster.demo.blog",
    "packageFolder": "com/jhipster/demo/blog",
    "reactive": true,
    "jhipsterVersion": "7.0.0-beta.0",
    "skipServer": false,
    "clientPackageManager": "npm",
    "dtoSuffix": "DTO",
    "entitySuffix": "",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "entities": ["Blog", "Post", "Tag"],
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "nativeLanguage": "en",
    "creationTimestamp": 1610419707891,
    "lastLiquibaseTimestamp": 1610419887000
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}
entity Post {
  title String required
  content TextBlob required
  date Instant required
}
entity Tag {
  name String required minlength(2)
}
relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}
relationship ManyToMany {
  Post{tag(name)} to Tag{entry}
}

paginate Post, Tag with infinite-scroll

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

git version 2.24.3 (Apple Git-128)

node: v14.15.0

npm: 6.14.11

yeoman: 3.1.1

Docker version 20.10.2, build 2291f61

docker-compose version 1.27.4, build 40524192

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #13466

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@qmonmert a suggestion is to bump minor dependencies and move every dependency to package.json and let dependabot bump majors versions.
And this will keep vue updated.

**Review by mraible — APPROVED** [review]

**Comment by mshima** [comment]
I am not sure about protractor dependencies.
They are not covered by main tests (only covered by daily build), they may require to bisect to find a problematic update.

WDYT?
Otherwise looks good.

File: generators/client/templates/vue/package.json [hunk 1]
```diff
@@ -1,19 +1,100 @@
...
+    "tslib": "2.1.0",
+    "tslint-config-prettier": "1.18.0",
+    "tslint-eslint-rules": "5.4.0",
+    "tslint-loader": "3.6.0",
+    "eslint": "7.17.0",
+    "eslint-plugin-prettier": "3.3.1",
+    "eslint-plugin-vue": "7.4.1",
     "typescript": "4.1.3",
+    "url-loader": "4.1.1",
+    "webdriver-manager": "12.1.7",
```
Comment (mraible) [inline generators/client/templates/vue/package.json:89]: This should be `"12.1.8"` to fix an issue [I found recently](https://github.com/jhipster/generator-jhipster/issues/13443#issuecomment-759215624).

**Comment by qmonmert** [comment]
@mshima & @mraible : I updated this PR with your returns

**Review by mshima — APPROVED** [review]

**Pull request merged**